bin/init.ml

```diff
@@ -38,15 +38,21 @@ let component_name_parser s =
     |> fun m -> `Msg m
   in
   let open Result.O in
-  let* atom = atom_parser s in
+  let* atom = atom_parser s ~err_msg:"expected a valid dune atom" in
   let* _ =
     match Lib_name.Local.of_string_opt s with
     | None -> Error (err_msg ())
     | Some s -> Ok s
   in
   Ok atom
 
-let atom_conv = Arg.conv (atom_parser, atom_printer)
+let ppx_name_conv =
```

Comment:
I'm not clear on the benefit of this addition. It doesn't add any new validation or checks, but seems to add code duplication and indirection just to add the names `ppx` and `lib`. More, these functions doesn't actually validate whether these are valid ppx or lib names, it just checks whether it's an atom.

Reply:
This was done to address your initial point of having a specific error message. Additionally, I think there is some atom parsing validation begin done. But I agree with you, it doesn't do much more than that and most just follows what is done for ```--ppx``` option. Should I remove this bit?

Reply:
Yeah, I think we're better off here just using `atom_conv`. "Is not a valid dune atom" should be a clear enough error message for the time being.

Reply:
This is done now.